File: Python/DSA/LinkedList/reverse_linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def reverse(head):
    prev = None
    curr = head

    while curr:
        next_node = curr.next
        curr.next = prev
        prev = curr
        curr = next_node

    return prev


# Create list manually
head = Node(1)
head.next = Node(2)
head.next.next = Node(3)

# Reverse
head = reverse(head)

# Print
temp = head
while temp:
    print(temp.data, end=" -> ")
    temp = temp.next

File: C/DSA/LinkedList/reverse_linked_list.c
#include <stdio.h>
#include <stdlib.h>


struct Node{

int data;
struct Node *next;

};



int main(){


struct Node *prev=NULL;
struct Node *current=NULL;
struct Node *next=NULL;


current=malloc(sizeof(struct Node));


current->data=10;


current->next=NULL;



while(current!=NULL){

next=current->next;

current->next=prev;

prev=current;

current=next;

}



printf("List Reversed");


return 0;

}